packages/engine/Source/Core/TileProviderError.js

```diff
@@ -138,9 +138,9 @@ TileProviderError.reportError = function (
     ++error.timesRetried;
   }
 
-  if (event.numberOfListeners > 0) {
+  if (defined(event) && event.numberOfListeners > 0) {
     event.raiseEvent(error);
-  } else {
+  } else if (defined(provider)) {
```

Comment:
Just to confirm, `provider` would always be set before and now it isn't? Or was this a bug? I see we made a lot of the parameters optional, which technically isn't a breaking change; but since these changed weren't mentioned in the PR description or in CHANGES, I figured I'd double check.

Reply:
Some of these (`message`, `x`, `y`, `level`) were already treated as optional, so the doc comment was updated for accuracy. 

`provider` and `event` were previously being passed, and this make it optional so that we can log errors without having already created the provider instance.